plugins/weather.py

```diff
@@ -0,0 +1,23 @@
+import urllib
+import json
+import ircbot.plugin
+
+class WeatherPlugin(ircbot.plugin.Plugin):
+	@ircbot.plugin.command('weather')
+	def weather(self, cmd):
+		if len(cmd.args) < 2:
+			return 'Usage: !weather city state/country'
+		try:
+			url = 'http://api.wunderground.com/api/ba6f8fecb3824c22/geolookup/conditions/q/' + cmd.args[1] + '/' + cmd.args[0] + '.json'
+			response = urllib.request.urlopen(url, timeout=2).read()
+			json_string = json.loads(response.decode('utf-8'))
+		except (urllib.error.URLError, urllib.error.HTTPError) as e:
+			return 'An error occured, try again later!'
+		try:
+			city = json_string['location']['city']
+			state = json_string['location']['state']
+			weather = json_string['current_observation']['weather']
+			temperature_string = json_string['current_observation']['temperature_string']
+		except KeyError as e:
+			return 'Usage: !weather city state/country'
```

Comment:
If this exception is raised, that would mean the JSON returned from the webservice is invalid - so this error message could probably be changed to "Invalid data returned" or something like that.